eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferCodecs.scala

```diff
@@ -40,9 +40,11 @@ object OfferCodecs {
   private val absoluteExpiry: Codec[AbsoluteExpiry] = tlvField(tu64overflow.as[TimestampSecond].as[AbsoluteExpiry])
 
   private val blindedNodeCodec: Codec[BlindedNode] = (("nodeId" | publicKey) :: ("encryptedData" | variableSizeBytes(uint16, bytes))).as[BlindedNode]
-
-  private val pathCodec: Codec[BlindedRoute] = (("firstNodeId" | publicKey) :: ("blinding" | publicKey) :: ("path" | listOfN(uint8, blindedNodeCodec).xmap[Seq[BlindedNode]](_.toSeq, _.toList))).as[BlindedRoute]
-
+  private val blindedNodesCodec: Codec[Seq[BlindedNode]] = listOfN(uint8, blindedNodeCodec).exmap(
+    nodes => if (nodes.length <= 1) Attempt.Failure(Err("blinded route must contain at least two nodes")) else Attempt.Successful(nodes.toSeq),
+    nodes => if (nodes.length <= 1) Attempt.Failure(Err("blinded route must contain at least two nodes")) else Attempt.Successful(nodes.toList),
```

Comment:
Why are you adding this artificial restriction?  Blinded routes with zero hop work just fine and are more efficient. Most big merchants are public and won't benefit from the increased privacy of a multi-hop blinded route and would prefer to use a minimal blinded route.

Reply:
That's a very good point. I was under the impression that we needed to use a dummy hop in such cases, but we actually don't. Rusty's recent comments made it look like cln doesn't handle a 0-hop blinded path, but we should ask them and update this depending on their answer.

Reply:
That was discussed during yesterday's spec meeting, and it was concluded that we cannot (and most likely should not) prevent recipients from revealing their identity if they want to, so we should allow 0-hop blinded routes. I'll update the PR to take that into account.

Reply:
I removed this restriction in the latest version.